clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/TopicMetadataApplicationEvent.java

```diff
@@ -20,15 +20,27 @@
 
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 
 public class TopicMetadataApplicationEvent extends CompletableApplicationEvent<Map<String, List<PartitionInfo>>> {
-    private final String topic;
-    public TopicMetadataApplicationEvent(final String topic) {
+
+    private final Optional<String> topic;
+
+    public TopicMetadataApplicationEvent(final Optional<String> topic) {
         super(Type.TOPIC_METADATA);
         this.topic = topic;
     }
 
-    public String topic() {
+    public Optional<String> topic() {
```

Comment:
I understand from the topic metadata request manager that empty here means the event would retrieve metadata for all topics right? I would suggest adding some documentation at the class level here to specify that the event has that behaviour

Reply:
@lianetm Take a look at the comment and LMK if it looks adequate. Thanks!